pkg/chunk/aws_storage_client.go

```diff
@@ -477,7 +477,7 @@ func (a awsStorageClient) getDynamoDBChunks(ctx context.Context, chunks []Chunk)
 	result := []Chunk{}
 	unprocessed := dynamoDBReadRequest{}
 	backoff, numRetries := minBackoff, 0
-	for outstanding.Len()+unprocessed.Len() > 0 {
+	for outstanding.Len()+unprocessed.Len() > 0 && numRetries < maxRetries {
```

Comment:
Yes, previously the ingester would try forever (but would return most of the time due to throughput errors usually being temporary). This is on the query path, so I'm unsure if anything is killed on request timeout.

No error is returned. Perhaps we should add one like we have in the write path? Does it make sense to error instead of returning a subset of the query results?